src/coreclr/System.Private.CoreLib/src/System/ValueType.cs

```diff
@@ -79,8 +95,44 @@ public override unsafe bool Equals([NotNullWhen(true)] object? obj)
         **Arguments: None.
         **Exceptions: None.
         ==============================================================================*/
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public extern override int GetHashCode();
+        public override unsafe int GetHashCode()
+        {
+            // The default implementation of GetHashCode() for all value types.
+            // Note that this implementation reveals the value of the fields.
+            // So if the value type contains any sensitive information it should
+            // implement its own GetHashCode().
+
+            MethodTable* pMT = RuntimeHelpers.GetMethodTable(this);
+
+            // We don't want to expose the method table pointer in the hash code
+            // Let's use the typeID instead.
+            uint typeID = RuntimeHelpers.GetTypeID(pMT);
```

Comment:
Is this something we still care about? I assume `GetType().GetHashCode()` would be slower.

Reply:
I think running the methodtable pointer through `HashCode` would be good enough. If you do that, the multiplication on the next won't be necessary.